llarp/router/outbound_session_maker.cpp

```diff
@@ -0,0 +1,293 @@
+#include <router/outbound_session_maker.hpp>
+
+#include <llarp/link/server.hpp>
+#include <util/logic.hpp>
+#include <router_contact.hpp>
+#include <nodedb.hpp>
+#include <router/i_rc_lookup_handler.hpp>
+#include <link/i_link_manager.hpp>
+
+namespace llarp
+{
+
+  struct PendingSession
+  {
+    //TODO: add session establish status metadata, e.g. num retries
+
+    const RouterContact rc;
+    LinkLayer_ptr link;
+
+    size_t attemptCount = 0;
+
+    PendingSession(const RouterContact &rc,
+                   LinkLayer_ptr link)
+        : rc(rc)
+        , link(link)
+    {
+    }
+
+  };
+
+  bool
+  OutboundSessionMaker::OnSessionEstablished(ILinkSession *session)
+  {
+    //TODO: do we want to keep it
+
+    const auto &router = session->GetPubKey();
+
+    LogInfo("session with ", router, " established");
+
+    if (not _rcLookup->RemoteIsAllowed(router))
```

Comment:
I actually prefer the `not` token over `!`

Reply:
Does MSVC support it yet (without `#include <ciso646>`)?

Reply:
Actually good question. I'm not sure. you just need `iso646.h` (since c99) but msvc is weird and is a c++ exclusive compiler 🤷‍♀

Reply:
update on this?  Also, perhaps `make format` can change to the preferred style?  Not sure, haven't really used it, but I imagine it'd be cool if it could.